'Serial' was not declared in this scope

Veryfing code using this library in Particle Build fails with this error:

```

```

sory forthe delay.
This was the error I get when compiling for Photon.

In file included from sd-card-library-photon-compat/sd2-card.h:26:0,
                 from sd-card-library-photon-compat/sd-fat-util.h:27,
                 from sd-card-library-photon-compat/sd-fat.h:27,
                 from sd-card-library-photon-compat/sd-card-library-photon-compat.h:18,
                 from spark-readwrite.cpp:23:
sd-card-library-photon-compat/sd2-card-config.h:21:0: warning: "Serial" redefined [enabled by default]
 #  define Serial Serial    /\* Use USB CDC port */
 ^
In file included from ./inc/application.h:45:0,
                 from spark-readwrite.cpp:22:
../wiring/inc/spark_wiring_usbserial.h:88:0: note: this is the location of the previous definition
 #define Serial _fetch_global_serial()
 ^
In file included from sd-card-library-photon-compat/sd2-card.h:26:0,
                 from sd-card-library-photon-compat/sd-fat-util.h:27,
                 from sd-card-library-photon-compat/sd-fat.h:27,
                 from sd-card-library-photon-compat/sd-card-library-photon-compat.h:18,
                 from spark-readwrite.cpp:23:
spark-readwrite.cpp: In function 'void setup()':
This looks like an error in sd-card-library-photon-compat library. Would you like to create an issue on GitHub to let the author know?

sd-card-library-photon-compat/sd2-card-config.h:21:18: error: 'Serial' was not declared in this scope
 #  define Serial Serial    /\* Use USB CDC port _/
                  ^
spark-readwrite.cpp:36:3: note: in expansion of macro 'Serial'
 const uint8_t misoPin = A4;
   ^
make[1]: *_\* [../build/target/user/platform-6spark-readwrite.o] Error 1
make: **\* [user] Error 2
Error: Could not compile. Please review your code.
Date: Mon, 14 Mar 2016 16:09:28 -0700
From: notifications@github.com
To: photon-sd-library@noreply.github.com
CC: kdupreez@hotmail.com
Subject: Re: [photon-sd-library] 'Serial' was not declared in this scope (#1)

Could you give a bit more information about the error please? Ideally the minimum amount of code required to reproduce and any errors you see

—
Reply to this email directly or view it on GitHub.

Hi,
I have been getting a similar error (below). The library and the code was working fine and one day I get the following error when I import and complied the SPARK-READWRITE.ino example from the SD-CARD-LIBRARY. Any suggestions? Thanks.

---

In file included from sd-card-library/sd2-card.h:26:0,
                 from sd-card-library/sd-fat-util.h:27,
                 from sd-card-library/sd-fat.h:27,
                 from sd-card-library/sd-card-library.h:18,
                 from spark-readwrite.cpp:23:
sd-card-library/sd2-card-config.h:21:0: warning: "Serial" redefined [enabled by default]
 #  define Serial Serial    /\* Use USB CDC port */
 ^
In file included from ./inc/application.h:45:0,
                 from spark-readwrite.cpp:22:
../wiring/inc/spark_wiring_usbserial.h:88:0: note: this is the location of the previous definition
 #define Serial _fetch_global_serial()
 ^
In file included from sd-card-library/sd2-card.h:26:0,
                 from sd-card-library/sd-fat-util.h:27,
                 from sd-card-library/sd-fat.h:27,
                 from sd-card-library/sd-card-library.h:18,
                 from spark-readwrite.cpp:23:
spark-readwrite.cpp: In function 'void setup()':

This looks like an error in sd-card-library library. Would you like to create an issue on GitHub to let the author know?

sd-card-library/sd2-card-config.h:21:18: error: 'Serial' was not declared in this scope
 #  define Serial Serial    /\* Use USB CDC port */
                  ^

spark-readwrite.cpp:36:3: note: in expansion of macro 'Serial'
 const uint8_t misoPin = A4;
   ^
make[1]: **\* [../build/target/user/platform-6spark-readwrite.o] Error 1
make: **\* [user] Error 2

Error: Could not compile. Please review your code.

--- a/firmware/Sd2Card_config.h
+++ b/firmware/Sd2Card_config.h
@@ -19,7 +19,7 @@
 /* Select output device for print() function */
 #	define SERIAL_DEVICE	0				/* 0:SerialUSB  1=: Serial UART device */
 #	if SERIAL_DEVICE == 0  	
-#		define Serial Serial				/* Use USB CDC port */
+//#		define Serial Serial				/* Use USB CDC port */
 #		define BPS_9600		/* Nothing */ 
 #		define BPS_115200 	/* Nothing */ 
 #	else

